### PR TITLE
Created library to interact with AWS IAM 

### PR DESCRIPTION
**Pull request by kelseylam**
Created library built off krux-boto that interacts with AWS IAM. Can create/delete/get users and keys and add/remove users from groups. Included a very basic CLI to test functionality.

### CODE REVIEW

File: krux_boto_iam/iam.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            AccessKeyId=key_id
+        )
+
+    def create_user(self, username):
+        """
+        Creates user and returns the user as a dict of their attributes.
+        If the user already exists returns none.
+        """
+        if self.get_user(username):
+            self._logger.error('AWS user with username: ' + username + ' already exists.')
```
Comment (hanpeter) [inline krux_boto_iam/iam.py]: Instead of an error log, throw an exception here. Let the caller decide what to do.

File: dev-requirements.pip [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# For unit tests
+# GOTCHA: Coverage is pegged at the latest version of 3, because of the error it throw when detemining
+# branch coverage.
+coverage==3.7.1
+# GOTCHA: This is the last version we can build in Jenkins due to the setuptools version limit.
+mock==1.1.2
+nose==1.3.7
+
+# Transitive Dependencies
+funcsigs==0.4
```
Comment (hanpeter) [inline dev-requirements.pip]: Nit: New line at the end of the file please.

**Comment by hanpeter** [comment]
You should leave a TODO for error handling (i.e. What happens if you delete a user that doesn't exist?)

Besides that, LGTM

File: requirements.pip [hunk 3]
```diff
@@ -0,0 +1,47 @@
+--extra-index-url https://staticfiles.krxd.net/foss/pypi/
+
+krux-stdlib==2.2.1
+krux-boto==1.0.3
```
Comment (hanpeter) [inline requirements.pip:3]: `krux-boto` has dependancies. Please add those. (Mostly `boto` and `boto3`)

**Comment by hanpeter** [comment]
Nit: Please also change the package name to `krux-iam`, just for consistency.

File: requirements.pip [hunk 4]
```diff
@@ -0,0 +1,25 @@
+--extra-index-url https://staticfiles.krxd.net/foss/pypi/
+
+krux-boto==1.0.3
+
```
Comment (hanpeter) [inline requirements.pip:4]: Nit: Please put following comment here. You can see the actual example [here](https://github.com/krux/python-krux-boto-s3/blob/master/requirements.pip).

```
# Transitive libraries
# This is needed so there are no version conflicts when
# one downstream library does NOT specify the version it wants,
# and another one does.
```

**Comment by hanpeter** [comment]
LGTM.

**Comment by dcrampton-sfdc** [comment]
Looks good, but you'll want more eyeballs on it.

File: krux_iam/iam.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        logger=None,
+        stats=None,
+    ):
+        # Private variables, not to be used outside this module
+        self._name = NAME
+        self._logger = logger or get_logger(self._name)
+        self._stats = stats or get_stats(prefix=self._name)
+
+        # Private client representing IAM
+        self._client = boto.client('iam')
```
Comment (arijitdey) [inline krux_iam/iam.py]: Nit- Would make 'iam' a constant.

File: krux_iam/iam.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+    def create_access_keys(self, username):
+        """
+        Creates and returns an AWS access key and secret key for the given user.
+        """
+        response = self._client.create_access_key(
+            UserName=username
+        )
+        key = response['AccessKey']
```
Comment (arijitdey) [inline krux_iam/iam.py:112]: What if the AccessKey is missing from response? Just minor checks.

**Comment by arijitdey** [comment]
Overall LGTM, apart from a couple of checks and nits.

**Pull request merged**